Algorithmia/client.py

```diff
@@ -141,7 +141,7 @@ def postJsonHelper(self, url, input_object, parse_response_as_json=True, **query
 
         response = self.requestSession.post(self.apiAddress + url, data=input_json, headers=headers, params=query_parameters)
 
-        if parse_response_as_json:
+        if parse_response_as_json and response.status_code == 200:
```

Comment:
checks  to ensure that the response is 200; which our API servers always set, but if there's a failure along the path we might not be able to deserialize as json